### PR TITLE
[16.0][MIG] sale_global_discount: 16.0

### PR DESCRIPTION
**Pull request by ferran-S73**
From https://github.com/OCA/sale-workflow/pull/2242

Needs https://github.com/OCA/account-invoicing/pull/1725, https://github.com/OCA/server-backend/pull/221

### CODE REVIEW

**Comment by rousseldenis** [comment]
@ferran-S73 Thanks for this. FYI, you can follow the following guide in order to integrate temporary requirements in order to be able to test it on runboat

https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

**Comment by ferran-S73** [comment]
> @ferran-S73 Thanks for this. FYI, you can follow the following guide in order to integrate temporary requirements in order to be able to test it on runboat
> 
> https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29

@rousseldenis done, thanks for the tip!

**Comment by ferran-S73** [comment]
@rousseldenis I'm not sure I'm doing it right, could you please check if the dependencies are correctly added?

**Comment by rousseldenis** [comment]
> @rousseldenis I'm not sure I'm doing it right, could you please check if the dependencies are correctly added?

@ferran-S73 It seems yes as runboat is started but tests are failing

**Comment by rousseldenis** [comment]
@ferran-S73

**Comment by ferran-S73** [comment]
> @ferran-S73

I haven't been able to figure out why my tests env isn't behaving like my local Odoo env. I seem to have a problem (in the tests only) in which instead of recalculating the tax line another one is added.
I'm gonna need more time to fix this issue

**Comment by BT-srieskamp** [comment]
Hi @ferran-S73 , I am curious about the current state of this migration. Do you see it possible to get it merged soon?

Kind Regards,
Sarah

**Comment by ferran-S73** [comment]
> Hi @ferran-S73 , I am curious about the current state of this migration. Do you see it possible to get it merged soon?
> 
> Kind Regards, Sarah

Hi @BT-skettler I'm having trouble fixing the tests for [account_global_discount ](https://github.com/OCA/account-invoicing/pull/1338) so it will still be a little until I merge them both. Sorry for the inconvenience!

**Review by QuocDuong1306 — APPROVED** [review]
LGTM

**Review by cyrilmanuel — APPROVED** [review]
not an expert but LGTM

**Review by leemannd — APPROVED** [review]

File: test-requirements.txt
Comment (cyrilmanuel) [inline test-requirements.txt]: having an error on :

Unreleased dependencies found in test-requirements.txt.

**Comment by miguel-S73** [comment]
I have fixed an error when `base_line` is `None`. 
https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move.py#L1199

**Comment by miguel-S73** [comment]
I have added this improvement:
https://github.com/OCA/sale-workflow/pull/2492

File: test-requirements.txt
Comment (GasparJarsa) [reply test-requirements.txt]: Could you please remove this file in order to pass the tests.

Comment (ferran-S73) [reply test-requirements.txt]: @GasparJarsa done

**Comment by QuocDuong1306** [comment]
Hello @ferran-S73 , maybe you need to update [test-requirements.txt](https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29) to check the tests

**Comment by ferran-S73** [comment]
> Hello @ferran-S73 , maybe you need to update [test-requirements.txt](https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference%28s%29-to-another-pull-request%28s%29) to check the tests

As suggested by other user https://github.com/OCA/sale-workflow/pull/2343#discussion_r1575265117 I did not update test-requirements

**Comment by QuocDuong1306** [comment]
I am not sure why he suggested that but
>having an error on :
Unreleased dependencies found in test-requirements.txt.

This error will occur when we have the PRs we need to depend on to pass the tests ( they will be automatically detected in `test-requirements.txt`), so don't worry about that error
After the dependent PRs are merged, we will remove the update `test-requirements.txt` commit and all will be ready to merge ( full green)

**Comment by QuocDuong1306** [comment]
Hi @ferran-S73 , I have [PR](https://github.com/Studio73/sale-workflow/pull/1) with:
- Small improvement
- Commit to update `test-requirement.txt`

It would be nice if you had a look

**Comment by trisdoan** [comment]
Hello @ferran-S73, I and @QuocDuong1306 are on the same team. We prepare another PR to include a fix in [here](https://github.com/Studio73/sale-workflow/pull/2)

Can you plz consider it?

**Comment by mariancuadranetkia** [comment]
Hello @ferran-S73!
Can you upgrade test-requirements and rebuild please? 
I think account_global_discount is now merged in 16.0 branch.
Thanks!

**Comment by mariancuadranetkia** [comment]
@ferran-S73 thanks! I tested it in runboat and LGTM :)
@rousseldenis do you think it still needs some revision?

**Review by alvaro-domatix — APPROVED** [review]

**Review by cyrilmanuel — APPROVED** [review]

**Comment by alvaro-domatix** [comment]
Hello @pedrobaeza , could you merge this PR when you have a moment? Thanks.

**Comment by pedrobaeza** [comment]
Merging following the reviews:

/ocabot merge nobump

**Pull request merged**
